mod_reforged/hooks/skills/effects/bleeding_effect.nut

```diff
@@ -0,0 +1,131 @@
+	q.m.MalusMultPerStack <- 0.05;
+
+	q.create = @(__original) function()
+	{
+		__original();
+		this.m.Description = "This character is bleeding profusely and will lose hitpoints each turn while acting slower and losing courage.";
+		this.m.IsStacking = false;
+	}
+
+	q.getTooltip <- function()
+	{
+		local ret = this.skill.getTooltip();
+		ret.push({
+			id = 10,
+			type = "text",
+			icon = "ui/icons/regular_damage.png",
+			text = ::Reforged.Mod.Tooltips.parseString("Receive " + ::MSU.Text.colorRed(this.getDamage()) + " damage per [turn|Concept.Turn]")
+		});
+		ret.push({
+			id = 11,
+			type = "text",
+			icon = "ui/icons/bravery.png",
+			text = ::Reforged.Mod.Tooltips.parseString(::MSU.Text.colorizeMult(this.getMalusMult()) + " less [Resolve|Concept.Bravery]")
+		});
+		ret.push({
+			id = 12,
+			type = "text",
+			icon = "ui/icons/initiative.png",
+			text = ::Reforged.Mod.Tooltips.parseString(::MSU.Text.colorizeMult(this.getMalusMult()) + " less [Initiative|Concept.Initiative]")
+		});
+		ret.push({
+			id = 13,
+			type = "text",
+			icon = "ui/icons/special.png",
+			text = ::Reforged.Mod.Tooltips.parseString("If at least " + ::MSU.Text.colorRed(this.m.StacksForMoraleCheck) + " stacks are received in a single [turn|Concept.Turn], a negative [morale check|Concept.Morale] is immediately triggered")
+		});
+		return ret;
+	}
+
+	q.getName <- function()
+	{
+		return this.m.Stacks == 1 ? this.m.Name : this.m.Name + " (x" + this.m.Stacks + ")";
+	}
+
+	q.getDescription = @() function()
+	{
+		return this.m.Description;
+	}
+
+	q.getMalusMult <- function()
+	{
+		return 1.0 - ::Math.minf(this.m.MaxMalusMult, this.m.Stacks * this.m.MalusMultPerStack);
+	}
+
+	q.getDamage <- function()
+	{
+		local ret = this.getContainer().hasSkill("trait.bleeder") ? this.m.Stacks * 2 : 1;
+		return this.getContainer().hasSkill("effects.hyena_potion") ? ::Math.floor(ret * 0.5) : ret;
+	}
+
+	// The vanilla file has an empty onUpdate function
+	q.onUpdate = @() function( _properties )
+	{
+		local mult = this.getMalusMult();
+		_properties.InitiativeMult *= mult;
+		_properties.BraveryMult *= mult;
+	}
+
+	// Based on the vanilla function but use this.getDamage() to calculate hitInfo.DamageRegular and use this.m.Overlay for spawnIcon
+	// and get rid of the part that decrements this.m.TurnsLeft and removes the effect
+	// also apply damage in instances smaller than ::Const.Morale.OnHitMinDamage to prevent morale check from this damage
+	q.applyDamage = @() function()
+	{
+		if (this.m.LastRoundApplied != ::Time.getRound())
+		{
+			this.m.LastRoundApplied = ::Time.getRound();
+			local actor = this.getContainer().getActor();
+			this.spawnIcon(this.m.Overlay, actor.getTile());
+
+			// Split the damage instances if damage is greater than ::Const.Morale.OnHitMinDamage
+			// so that the damage does not trigger morale checks
+			local damage = [this.getDamage()];
+			local i = 0;
+			while (i < damage.len() && damage[i] > ::Const.Morale.OnHitMinDamage)
```

Comment:
You do this in order to not cause morale hits by bleed damage. But this does not achieve that yet. If the bleeding does 15 damage then that is **not** greater than the vanilla `OnHitMinDamage` because it is equal to that. So the bleeding damage will not be split. And the 15 damage will later cause a morale check.

Once you fixed that, then a `OnHitMinDamage` of 1 will now also cause an infinite loop